src/role_carry.js

```diff
@@ -55,15 +55,15 @@ function checkHelperEmptyStorage(creep) {
   if (creep.room.name === creep.memory.routing.targetRoom) {
     const targetStructure = Game.getObjectById(creep.memory.routing.targetId);
     if (targetStructure === null) {
-      creep.suicide();
+      Creep.recycleCreep(creep);
       return;
     }
 
     if (targetStructure.structureType === STRUCTURE_STORAGE) {
       creep.say('storage');
       if (targetStructure.store.energy === 0) {
         creep.log('Suiciding the storage I should get the energy from is empty');
```

Comment:
this should work 
but `creep.log('recycling: the storage I should get the energy from is empty');`